### PR TITLE
Fix multistrain Rt

### PR DESCRIPTION
**Pull request by edknock**
For the multistrain version of the model, `Rt` is calculated using `prob_strain` - which is essentially the proportion of force of infection that is the non-variant strain. However, if there is no force of infection then we yield `NA` values for `prob_strain`. This resulted in the eigenvalue calculation within the `Rt` calculation throwing an error.

This PR fixes this by setting `Rt` to `NA` in cases where `prob_strain` is `NA`.

### CODE REVIEW

File: R/carehomes_rt.R [hunk 1]
```diff
@@ -80,25 +80,30 @@ carehomes_Rt <- function(step, S, p, prob_strain = NULL) {
...
+        i_CHR <- seq(p$n_groups, nrow(ngm), by = p$n_groups)
+        i_CHW <- i_CHR - 1
+        i_gen <- seq_len(nrow(ngm))[-c(i_CHW, i_CHR)]
+        ngm <- ngm[i_gen, i_gen]
+      }
+      ev <- eigen(ngm)$values
+      ev[Im(ev) != 0] <- NA
+      ev <- as.numeric(ev)
+      out <- max(ev, na.rm = TRUE)
     }
```
Comment (richfitz) [inline R/carehomes_rt.R:106]:
```suggestion
      out <- max(ev, na.rm = TRUE)
    }
    out
```

Comment (richfitz) [reply R/carehomes_rt.R:106]: this makes the return more explicit (currently the upper branch of the `if` is a long way from the bottom of the function)

**Review by richfitz — CHANGES_REQUESTED** [review]
one tiny tweak here, but this seems reasonable

File: R/carehomes_rt.R (hunk 1 above)
Comment (edknock) [reply R/carehomes_rt.R:106]: I meant to do that but clearly missed it. Thanks for catching that.

**Review by richfitz — APPROVED** [review]

**Pull request merged**